tests/transformer/test_async.py

```diff
@@ -0,0 +1,157 @@
+from RestrictedPython import compile_restricted_exec
+from RestrictedPython._compat import IS_PY35_OR_GREATER
+from RestrictedPython.transformer import RestrictingNodeTransformer
+from tests import c_exec
+
+import pytest
+
+pytestmark = pytest.mark.skipif(
+    not IS_PY35_OR_GREATER,
+    reason="async statement was first introduced in Python 3")
+
+
+# Example from https://docs.python.org/3/library/asyncio-task.html
+ASYNC_DEF_EXMAPLE = """
+import asyncio
+
+async def hello_world():
+    print()
+
+loop = asyncio.get_event_loop()
+# Blocking call which returns when the hello_world() coroutine is done
+loop.run_until_complete(hello_world())
+loop.close()
+"""
+
+
+@pytest.mark.parametrize(*c_exec)
+def test_async_def(c_exec):
+    result = c_exec(ASYNC_DEF_EXMAPLE)
+    assert result.code is None
+    assert result.errors == (
+        'Line 4: AsyncFunctionDef statements are not allowed.',
+    )
+    assert result.warnings == []
+    assert result.used_names == {
+        'asyncio': True,
+        'hello_world': True,
+        'loop': True,
+    }
+
+
+# Modified Example from http://stackabuse.com/python-async-await-tutorial/
+YIELD_FORM_EXAMPLE = """
+import asyncio
+
+@asyncio.coroutine
+def get_json(client, url):
+    file_content = yield from load_file('data.ini')
+"""
+
+
+@pytest.mark.parametrize(*c_exec)
+def test_yield_from(c_exec):
+    result = c_exec(YIELD_FORM_EXAMPLE)
+    assert result.code is None
+    assert result.errors == (
+        'Line 6: YieldFrom statements are not allowed.',
+    )
+    assert result.warnings == []
+    assert result.used_names == {
+        'asyncio': True,
+    }
+
+# special efford to test await, async for and async with
+
+
+class RestrictingAsyncNodeTransformer(RestrictingNodeTransformer):
+
+    def visit_AsyncFunctionDef(self, node):
+        """
+        AsyncFunctionDef needs to be allowed for await,
+        async for and async with
+        """
+        return self.node_contents_visit(node)
+
+
+# Modified example from https://docs.python.org/3/library/asyncio-task.html
+AWAIT_EXAMPLE = """
+import asyncio
+import datetime
+
+async def display_date(loop):
+    end_time = loop.time() + 5.0
+    while True:
+        print(datetime.datetime.now())
+        if (loop.time() + 1.0) >= end_time:
+            break
+        await asyncio.sleep(1)
+
+loop = asyncio.get_event_loop()
+# Blocking call which returns when the display_date() coroutine is done
+loop.run_until_complete(display_date(loop))
+loop.close()
+"""
+
+
+@pytest.mark.parametrize(*c_exec)
+def test_await(c_exec):
+    result = compile_restricted_exec(
+        AWAIT_EXAMPLE,
+        policy=RestrictingAsyncNodeTransformer)
+    assert result.code is None
+    assert result.errors == (
+        'Line 11: Await statements are not allowed.',
+    )
+    assert result.warnings == [
```

Comment:
I see no reason for this line and the following ones in this function as its only goal is to show that `async` functionality is not allowed.